asv/benchmark.py

```diff
@@ -718,6 +719,40 @@ def run(self, *param):
         return sizeofcopy - sizeof2
 
 
+class RayMemBenchmark(Benchmark):
+    """
+    A single benchmark for tracking the memory of an object with memray
+    """
+
+    name_regex = re.compile("^(Ray[A-Z_].+)|(ray_.+)$")
+
+    def __init__(self, name, func, attr_sources):
+        Benchmark.__init__(self, name, func, attr_sources)
+        self.type = "memory"
+        self.unit = "bytes"
+        pass
+
+    def run(self, *param):
+        try:
+            import memray
+            from memray import FileReader
+        except ImportError:
+            subprocess.check_output([sys.executable, "-mpip", "install", "memray"])
+            import memray
+            from memray import FileReader
```

Comment:
We need to add `memray` as a runtime or optional dependency. Running `pip install` during runtime creates an undesired side effect.

Reply:
This was implemented in this manner so `memray` is part of each environment, since `asv` doesn't allow importing modules (including `asv` itself) within this `benchmark.py` file.

Reply:
> This was implemented in this manner so `memray` is part of each environment, since `asv` doesn't allow importing modules (including `asv` itself) within this `benchmark.py` file.

I guess this was the reason `asv` initial vendored the `sizeof` module.

Reply:
Yup, but it wasn't a very flexible design. The `pytest` approach to this (collection / isolation) is probably something worth looking into more.